exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/PrometheusHttpServerBuilder.java

```diff
@@ -32,13 +38,30 @@ public PrometheusHttpServerBuilder setPort(int port) {
     return this;
   }
 
+  /** Sets the {@link ExecutorService} to be used for {@link PrometheusHttpServer}. */
+  public PrometheusHttpServerBuilder setExecutor(ExecutorService executor) {
+    requireNonNull(executor, "executor");
+    this.executor = executor;
+    return this;
+  }
+
   /**
    * Returns a new {@link PrometheusHttpServer} with the configuration of this builder which can be
    * registered with a {@link io.opentelemetry.sdk.metrics.SdkMeterProvider}.
    */
   public PrometheusHttpServer build() {
-    return new PrometheusHttpServer(host, port);
+    ExecutorService executorService = this.executor;
+    if (executorService == null) {
```

Comment:
The `executorService` cannot be null here; you're already setting it in the default constructor. Consider removing the assignment in the constructor instead